src/main/java/io/quarkus/search/app/SearchService.java

```diff
@@ -68,6 +73,22 @@ public SearchResult<SearchHit> search(@RestQuery @DefaultValue(QuarkusVersions.L
                                         .boost(50.0f)));
                     }
                 })
+                // * Highlighters are going to use spans-with-classes so that we will have more control over styling the visual on the search results screen.
+                // * We give control to the caller on the content snippet length and the number of these fragments
+                // * No match size is there to make sure that we are still going to get the text even if the field didn't have a match in it.
+                // * The title in the Guide entity is `Length.LONG` long, so we use that as a max value for no-match size, but hopefully nobody writes a title that long...
+                .highlighter(
+                        f -> f.unified().noMatchSize(Length.LONG).fragmentSize(0)
+                                .orderByScore(true)
+                                // just in case we have any "unsafe" content:
+                                .encoder(HighlighterEncoder.HTML)
+                                .numberOfFragments(1)
+                                .tag("<span class=\"" + highlightCssClass + "\">", "</span>")
+                                .boundaryScanner().sentence().end())
+                // * If there's no match in the full content we don't want to return anything.
+                // * Also content is really huge, so we want to only get small parts of the sentences. We are allowing caller to pick the number of sentences and their length:
+                .highlighter("highlighter_content",
+                        f -> f.unified().noMatchSize(0).numberOfFragments(contentSnippets).fragmentSize(contentSnippetsLength))
```

Comment:
I'm surprise this doesn't fail hard with the default values of 0?

Reply:
Highlights work in mysterious ways 😄 
I've changed the defaults a bit... I was expecting that with 0 `numberOfFragments`  it won't return anything, but turns out it returns all the content 🙈 
If we'd wanted to not return that part at all, then we'd need to have a different projection and pass it in conditionally... Do you think it's worth having that?

Reply:
> Do you think it's worth having that?

Not necessarily, but I think it's worth having defaults that work properly (e.g. `numberOfFragments = 5`) and to throw a proper exception for invalid input (feel free to add Hibernate Validator to the project to validate method parameters).

Reply:
> Not necessarily, but I think it's worth having defaults that work properly (e.g. numberOfFragments = 5)

Ok, seems you did that part. LGTM.

> and to throw a proper exception for invalid input (feel free to add Hibernate Validator to the project to validate method parameters).

This is not really necessary, I was getting ahead of myself. Please disregard.